.github/workflows/main.yml

```diff
@@ -106,18 +121,20 @@ jobs:
     name: Build Clusters on JDK ${{ matrix.java }}
     if: contains(github.event.pull_request.labels.*.name, 'ci:no-build') == false
     runs-on: ubuntu-latest
-    timeout-minutes: 60
+    timeout-minutes: 40
     strategy:
       matrix:
         java: [ '17', '21', '22' ]
+        exclude:
+          - java: ${{ github.event_name == 'pull_request' && 'nothing' || '21' }}
```

Comment:
for future reference: this trick allows us to reduce the test matrix using labels or other properties